crates/diman_unit_system/src/codegen/dimension_def.rs

```diff
@@ -102,19 +102,22 @@ impl Defs {
                     }
                 }
 
+                #[cfg(any(feature = "std", feature = "float_core"))]
                 pub const fn dimension_powi(self, other: i32) -> Self {
                     Self {
                         #powi_gen
                     }
                 }
 
+                #[cfg(any(feature = "std", feature = "float"))]
```

Comment:
> Also, if dimension_sqrt is only ever evaluated at compile-time (where we have std), then technically don't need it at run-time (a no_std environment). I don't know how we would express that, but it seems possible.

Oh, is that possible? It seems reasonable, but I don't know whether that is supported by rustc. 

I take it the part that doesn't compile in these methods is the `panic!`, right? Unfortunately I don't really see a solution for #42 (other than the user using rational dimensions).

Reply:
> This is a bad approach because it's evaluated in the context of diman, not diman_unit_system. I can rework this to get the #[cfg] outside the quote! block.

Keep in mind that if the user defines their own `unit_system!` call, then it would be evaluated in the context of the users crate, so feature gates will always have to be extracted out of the generated code.

Reply:
> Oh, is that possible? It seems reasonable, but I don't know whether that is supported by rustc.

All I know of are macro-based compile-time evaluators, which seem too hacky to pursue.

> I take it the part that doesn't compile in these methods is the `panic!`, right?

Hmm, actually it looks like I can remove these guards. I placed them early because `sqrt` and `cbrt` are not available in `no_std` (without "num-traits/libm"), but it seems whatever problem I had is no longer relevant (and the implementation doesn't depend on those numerical functions).